tests/integration_toolbox.py

```diff
@@ -156,6 +158,46 @@ def testGetUrlToFileCachedKeepaliveChanged(self):
             conn.close()
         self.assertEqual(b'This is a test\n', data)
 
+    def testGetUrl404(self):
+        """Call get_url for a file that doesn't exist"""
+        with self.assertRaises(HTTPError):
+            data = spacepy.toolbox.get_url(
+                'http://localhost:{}/doesnotexist.txt'.format(self.port))
+    
+    def testGetUrlNoConnNoKeepAlive(self):
+        """Call get_url without specifying connection or keepalive"""
+        ctype = http.client.HTTPConnection
+        connection = ctype("localhost:{}".format(self.port))
```

Comment:
There's no point in the assignment since it's not used again (there were some cases where we were doing this with a different assignment for Python 2 and 3 where the class name changed, but not relevant here)--so this can just be one line, in all the cases.

The connection should also be closed at the end, which basically means putting everything from the point that the connection is made to the end within a try/finally block. True for all these tests.

Reply:
I'm not completely sure what you meant by this, so,I got rid of the connection variable and called the ctype() in the get_url function arguments

Reply:
The connection assignment is useful (particularly since you have to close it later). The ctype one isn't...just instantiate http.client.HTTPConnection directly.